### PR TITLE
update sharing test to req aws account id

### PR DESCRIPTION
**Pull request by krantzinator**
Replaces the hardcoded AWS account ID with an environment variable. Includes a precheck for the env var's existence.

Test results with the unset env var:
```
=== RUN   TestBuilderAcc_amiSharing
    TestBuilderAcc_amiSharing: builder_acc_test.go:254: TESTACC_AWS_ACCOUNT_ID must be set for acceptance tests
--- FAIL: TestBuilderAcc_amiSharing (0.00s)
```

Test results with the set env var:
```
--- PASS: TestBuilderAcc_amiSharing (157.03s)
PASS
ok      github.com/hashicorp/packer/builder/amazon/ebs  157.042s
```

### CODE REVIEW

File: builder/amazon/ebs/builder_acc_test.go [hunk 1]
```diff
@@ -351,3 +358,7 @@ func buildForceDeregisterConfig(val, name string) string {
 func buildForceDeleteSnapshotConfig(val, name string) string {
 	return fmt.Sprintf(testBuilderAccForceDeleteSnapshot, val, val, name)
 }
+
+func buildSharingConfig(val string) string {
```
Comment (sylviamoss) [inline builder/amazon/ebs/builder_acc_test.go:362]: Nice!!

**Review by sylviamoss — APPROVED** [review]
LGTM!!

**Review by nywilken — APPROVED** [review]
This looks good.

**Comment by catsby** [comment]
_Finally_ 😉

**Pull request merged**